Library/Homebrew/formula.rb

```diff
@@ -1519,10 +1519,15 @@ def recursive_requirements(&block)
   # Returns a list of Dependency objects that are required at runtime.
   # @private
   def runtime_dependencies
-    recursive_dependencies do |_dependent, dependency|
+    runtime_dependencies = recursive_dependencies do |_dependent, dependency|
       Dependency.prune if dependency.build?
       Dependency.prune if !dependency.required? && build.without?(dependency)
     end
+    runtime_dependencies += recursive_requirements do |_dependent, requirement|
+      Requirement.prune if requirement.build?
+      Requirement.prune if !requirement.required? && build.without?(requirement)
+    end.map(&:to_dependency).compact
+    runtime_dependencies
```

Comment:
Yeah, I agree. Warning when uninstalling is a nice-to-have but not an essential. (I'm still not 100% certain any sort of fallback is necessary.)